### PR TITLE
fix: Handle commas in permissions array

### PR DESCRIPTION
**Pull request by erezrokah**
#### Summary

Follow up to https://github.com/cloudquery/plugin-sdk/pull/1827#discussion_r1696683944

---

Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `go fmt` to format your code 🖊
- [ ] Lint your changes via `golangci-lint run` 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Update or add tests 🧪
- [ ] Ensure the status checks below are successful ✅

### CODE REVIEW

File: schema/table.go [hunk 1]
```diff
@@ -182,14 +182,17 @@ func NewTableFromArrowSchema(sc *arrow.Schema) (*Table, error) {
 	for i, field := range fields {
 		columns[i] = NewColumnFromArrowField(field)
 	}
+
+	var permissionsNeededArr []string
+	_ = json.Unmarshal([]byte(permissionsNeeded), &permissionsNeededArr)
```
Comment (erezrokah) [inline schema/table.go:187]: Don't think `Marshal`, `Unmarshal` can fail for `[]string` and if it does I don't think it should panic for a doc related feature

Comment (erezrokah) [reply schema/table.go:187]: I mean it can fail if the string is not a JSON array but we're the ones setting it

Comment (marianogappa) [reply schema/table.go:187]: Can it fail if it's not set?

Comment (marianogappa) [reply schema/table.go:187]: Is that a good test case?

Comment (erezrokah) [reply schema/table.go:187]: If it's not set `nil` array gets serialized to `null` I'll add a test for it

Comment (erezrokah) [reply schema/table.go:187]: Added https://github.com/cloudquery/plugin-sdk/pull/1835/commits/baf5f303c65d2f9680c0641248581cacc78c3144

`nil` columns are always recreated as non `nil` due to https://github.com/cloudquery/plugin-sdk/blob/341c54641fc2de65c23287276989d9aebca2aecd/schema/table.go#L181

**Review by marianogappa — APPROVED** [review]

**Pull request merged**